### PR TITLE
Tsc 389 column menu scroll up on navigation bar

### PR DESCRIPTION
**Pull request by Changk820**
image
That blue button only shows when scroll down a bit, roughly when 5 rows are scrolled pass it will show. 

I also tested to make sure if I expand the menu to be big and keep scrolling, the position of the button is always fixed.

There is a slight issue though, as you can see from the CSS, I had to do some weird hard coding of its position. I tried doing absolution position, but no matter what, the position is in relation to the full window of the site, not  the box the setting names are in, thus I ended up just using position: fixed and percentages to get it right inside the box.

If you have any ideas, lmk.

For now, I'm going to make pull request and deploy to see if it looks correct on 27 in.

### CODE REVIEW

**Comment by paologum** [comment]
the arrow should be within the accordion box

image

**Comment by Changk820** [comment]
not done yet btw

**Comment by Changk820** [comment]
> the arrow should be within the accordion box
> 
> image

How were you able to get that image?
I think it works now because I zoomed in a lot and it still stays in, same with using inspect to preview it. I couldn't get deploy to work

image

image

image

**Comment by Changk820** [comment]
I cant get it to fade in fade out, but the other changes you asked for are there

**Comment by paologum** [comment]
to get it to fade in and out, look at transitions with css and change visibilities/display when it appears vs when it doesn't appear

**Comment by paologum** [comment]
there is code for that again in the map points legend scroll to top button

**Comment by paologum** [comment]
hmm wait looking at the code give me a second, bc you did use the map points legend thingie

**Review by paologum — CHANGES_REQUESTED** [review]

File: app/src/settings_tabs/Column.tsx [hunk 1]
```diff
@@ -61,6 +92,17 @@ export const Column = observer(function Column() {
             Object.entries(state.settingsTabs.columns.children).map(([childName, childDetails]) => (
               <ColumnAccordion key={childName} details={childDetails} />
             ))}
+          {/* Button to take users to top of column menu when scrolling */}
+          {showScroll && (
+            <IconButton onClick={scrollToTop} className={`scroll-to-top-button ${showScroll ? "show" : ""}`}>
+              <Lottie
+                key="settings-arrow-up"
+                style={{ width: "28px", height: "28px" }}
+                animationData={LightArrowUpIcon}
```
[the remark this answers is not in the document]
Comment (Changk820) [reply app/src/settings_tabs/Column.tsx:101]: I thought the darkarrow was kind of ugly
image
image


Should it maybe be swapped?
image
image

to match the check mark colors

File: app/src/settings_tabs/Column.css [hunk 2]
```diff
@@ -261,3 +261,41 @@
     font-size: 0.8rem;
   }
 }
+
+.scroll-to-top-button {
+  position: sticky;
+  bottom: 10px;
+  left: 90%;
+  background-color: #6693c9;
+  color: #fff;
```
[the remark this answers is not in the document]
Comment (Changk820) [reply app/src/settings_tabs/Column.css:270]: i applied the brightness thing in next step, without background you don't see it light up as well, but the button i guess is self explanatory anyways
image

File: app/src/settings_tabs/Column.tsx [hunk 3]
```diff
@@ -61,6 +94,17 @@ export const Column = observer(function Column() {
             Object.entries(state.settingsTabs.columns.children).map(([childName, childDetails]) => (
               <ColumnAccordion key={childName} details={childDetails} />
             ))}
+          {/* Button to take users to top of column menu when scrolling */}
+
+          <IconButton onClick={scrollToTop} className={`scroll-to-top-button ${showScroll ? "show" : ""}`}>
+            <Lottie
+              key="settings-arrow-up"
+              style={{ width: "28px", height: "28px" }}
+              // animationData={theme.palette.mode === "light" ? LightArrowUpIcon : DarkArrowUpIcon}
```
Comment (paologum) [inline app/src/settings_tabs/Column.tsx:103]: just remove this comment and you should be good

**Review by paologum — APPROVED** [review]

File: app/src/settings_tabs/Column.tsx (hunk 3 above)
Comment (Changk820) [reply app/src/settings_tabs/Column.tsx:103]: I forgot to remove while testing appearance  lol.

**Pull request merged**